hantec_api_ecommerce/controllers/main.py

```diff
@@ -0,0 +1,915 @@
+from odoo.http import request, Controller, route
+import logging, re
+
+logger = logging.getLogger(__name__)
+
+
+class MainController(Controller):
+    @route("/create_contact", methods=["POST"], type="json", auth="user")
+    def create_contact(self):
+        """
+        Combines the logic of searching and creating contacts based on email, phone, or store name.
+
+        This function first checks if a contact already exists based on the provided
+        email, phone, or store name. If a matching contact is found, it returns the contact ID.
+        If no matching contact is found, it creates a new contact with the provided data.
+
+        JSON request body:
+            - email (str, optional): The email of the contact.
+            - phone (str, optional): The phone number of the contact.
+            - store_name (str, optional): The store name associated with the contact.
+            - partner_id (int, optional): The parent ID for the contact.
+            - contact_data (dict, optional): Additional contact data.
+
+        JSON response:
+            - message (str): A message indicating whether a contact was found or created.
+            - contact_id (int): The ID of the found or created contact.
+
+        Returns:
+            dict: A dictionary with a message and the contact ID.
+
+        """
+        env = request.env
+        data = request.jsonrequest
+        email = data.get("email")
+        phone = data.get("phone")
+        store_name = data.get("store_name")
+        partner_id = data.get("partner_id")
+        contact_data = data.get("contact_data", {})
+
+        if email or phone:
+            email_pattern = r"^([a-z0-9!#$%&'*+/=?^_`{|}~-]+(?:\.[a-z0-9!#$%&'*+/=?^_`{|}~-]+)*)"  # Email pattern to get the value before @
+            email_prefix = re.match(email_pattern, email).group() if email else None
+            phone_suffix = phone[len(phone) - 4 :] if phone else None
+
+            domain = []
+            if email_prefix:
+                domain.append(("email", "ilike", f"{email_prefix}%"))
+            if phone_suffix:
+                domain.append(("phone", "ilike", f"%{phone_suffix}"))
+
+            existing_contact = env["res.partner"].search(domain, limit=1)
+
+            if existing_contact:
+                logger.info("Contact found with ID %s", existing_contact.id)
+                return {
+                    "message": f"Contact found with ID: {existing_contact.id}.",
+                    "contact_id": existing_contact.id,
+                }
+
+            if "email" not in contact_data and email:
+                contact_data["email"] = email
+            if "phone" not in contact_data and phone:
+                contact_data["phone"] = phone
+
+        if store_name:
+            store_name_suffix = store_name[len(store_name) - 4 :]
+
+            domain = [("name", "ilike", f"%{store_name_suffix}")]
```

Comment:
```suggestion
            domain = [("name", "=", f"%{store_name_suffix}")]
```
why not to use = operator?